tests/smoke_tests/run_smoke_test.py

```diff
@@ -0,0 +1,182 @@
+import asyncio
+import logging
+from pathlib import Path
+
+import torch
+import yaml
+from flwr.common.typing import Config
+from six.moves import urllib
+
+from examples.fedprox_example.client import MnistFedProxClient
+from fl4health.utils.metrics import Accuracy
+
+logging.basicConfig(format="%(asctime)s %(levelname)-8s %(message)s", level=logging.INFO, datefmt="%Y-%m-%d %H:%M:%S")
+logger = logging.getLogger()
+
+
+async def run_smoke_test(
+    config_path: str = "tests/smoke_tests/config.yaml",
+    dataset_path: str = "examples/datasets/mnist_data/",
+) -> None:
+    with open(config_path, "r") as file:
+        config = yaml.safe_load(file)
+
+    _preload_dataset(dataset_path, config)
+
+    # Start the server and capture its process object
+    logger.info("Starting server...")
+    server_process = await asyncio.create_subprocess_exec(
+        "nohup",
+        "python",
+        "-m",
+        "examples.fedprox_example.server",
+        "--config_path",
+        config_path,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.STDOUT,
+    )
+
+    # reads lines from the server output in search of the startup log message
+    # times out after 20s of inactivity if it doesn't find the log message
+    full_server_output = ""
+    startup_message = "FL starting"
+    output_found = False
+    while True:
```

Comment:
Super minor, but it might be slightly cleaner to do
```
while not output_found:
    ...
    if startup_message in server_output:
        output_found = True
```
avoiding the "break"